lib/index.js

```diff
@@ -404,11 +410,14 @@ exports.init = function(options, callback) {
         var restricted = filtered;
 
         // package whitelist
-        if (options.packages) {
-            var packages = options.packages.split(';');
+        if (packages) {
             restricted = {};
             Object.keys(filtered).map(function(key) {
-                if (packages.includes(key)) {
+                // Whitelist packages by declaring:
+                // 1. the package full name. Ex: `react` (we suffix an '@' to ensure we don't match packages like `react-native`)
+                // 2. the package full name and the major version. Ex: `react@16`
+                // 3. the package full name and full version. Ex: `react@16.0.2`
+                if (packages.findIndex((package) => key.includes(package.includes('@') ? package : package + '@')) !== -1) {
```

Comment:
I found two minor issues with your changes:
1. If I specify `yup`, it will also match `@types/yup`
2. If I specify `@material-ui/core`, it will not match `@material-ui/core@1.2.3`

I fixed (1) by using `startsWith` and (2) by checking that the `@` is present and not the first character `.indexOf("@") > 0` 
```suggestion
                if (packages.findIndex((package) => key.startsWith(package.indexOf('@') > 0 ? package : package + '@')) !== -1) {
```

Reply:
Nice finding @cmfcmf! 

Although, I'm gonna wrap that logic and still add the `@` for both cases so you can specify the right packages.

Let's say that I've passed down `@babel/preset-a`, but for some reason, there's also `@babel/preset-agnostic`, both would match with this logic.